packages/grafana-ui/src/components/Combobox/useComboboxFloat.ts

```diff
@@ -48,46 +48,25 @@ export const useComboboxFloat = (
     whileElementsMounted: autoUpdate,
   });
 
-  useEffect(() => {
-    if (range === null) {
-      return;
-    }
-    const startVisibleIndex = range?.startIndex;
-    const endVisibleIndex = range?.endIndex;
+  const longestItemWidth = useMemo(() => {
+    let longestItem = '';
+    const itemsToLookAt = Math.min(items.length, WIDTH_CALCULATION_LIMIT_ITEMS);
 
-    if (typeof startVisibleIndex === 'undefined' || typeof endVisibleIndex === 'undefined') {
-      return;
+    for (let i = 0; i < itemsToLookAt; i++) {
+      const itemLabel = items[i].label;
+      longestItem = itemLabel.length > longestItem.length ? itemLabel : longestItem;
     }
 
-    // Scroll down and default case
-    if (
-      startVisibleIndex === 0 ||
-      (startVisibleIndex % INDEX_WIDTH_CALCULATION === 0 && startVisibleIndex >= INDEX_WIDTH_CALCULATION)
-    ) {
-      let maxLength = 0;
-      const calculationEnd = Math.min(items.length, endVisibleIndex + INDEX_WIDTH_CALCULATION);
-
-      for (let i = startVisibleIndex; i < calculationEnd; i++) {
-        maxLength = Math.max(maxLength, items[i].label.length);
-      }
+    // pr todo: get weight from theme/styles
+    const size = measureText(longestItem, MENU_ITEM_FONT_SIZE, 500).width;
 
-      setPopoverWidth(maxLength * WIDTH_MULTIPLIER);
-    } else if (endVisibleIndex % INDEX_WIDTH_CALCULATION === 0 && endVisibleIndex >= INDEX_WIDTH_CALCULATION) {
-      // Scroll up case
-      let maxLength = 0;
-      const calculationStart = Math.max(0, startVisibleIndex - INDEX_WIDTH_CALCULATION);
-
-      for (let i = calculationStart; i < endVisibleIndex; i++) {
-        maxLength = Math.max(maxLength, items[i].label.length);
-      }
-
-      setPopoverWidth(maxLength * WIDTH_MULTIPLIER);
-    }
-  }, [items, range, setPopoverWidth]);
+    // pr todo: ideally we want to remove these magic numbers from here, or derive them from the styles
```

Comment:
Can we do some kind of measurement here?

Reply:
what?

Reply:
@tskarhed what?

Reply:
Well it is either taking it from the styles (which might be preferred) or doing an actual measuring of the element (without the text)?